### PR TITLE
[prometheus-stackdriver-exporter] fix deployment annotations

### LINKED ISSUE #4768: [prometheus-stackdriver-exporter] Deployment metadata missing annotations
### Describe the bug a clear and concise description of what the bug is.

when in values annotations are defined, they are not loaded into Deployment

### What's your helm version?

version.BuildInfo{Version:"v3.15.2", GitCommit:"1a500d5625419a524fdae4b33de351cc4f58ec35", GitTreeState:"clean", GoVersion:"go1.22.4"}

### What's your kubectl version?

Client Version: v1.29.5

### Which chart?

prometheus-stackdriver-exporter

### What's the chart version?

4.6.0

### What happened?

Deployment doesn't contains annotations defined in values

### What you expected to happen?

add anotation in deployment metadata:

apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-prometheus-stackdriver-exporter
  annotations:
    

### How to reproduce it?

just install helm and Deployment dosn't have annotations

### Enter the changed values of values.yaml?

annotations:
  sample: value

### Enter the command that you execute and failing/misfunctioning.

helm template test prometheus-community/prometheus-stackdriver-exporter -f ../terraform/helm/prometheus-stackdriver-exporter/values.yaml

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by wsmolkowski**
#### What this PR does / why we need it
fix #4768
#### Which issue this PR fixes

- fixes #4768 

#### Checklist

- [X] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by rpahli — APPROVED** [review]

**Comment by wsmolkowski** [comment]
@rpahli when could you merge it?

**Comment by rpahli** [comment]
@wsmolkowski there are still build error. Can you pls fix the build, then we can merge your PR.

**Comment by rpahli** [comment]
@wsmolkowski any update on this? can you pls fix the lint error?

File: charts/prometheus-stackdriver-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ metadata:
   name: {{ template "stackdriver-exporter.fullname" . }}
   labels:
     {{- include "stackdriver-exporter.labels" . | indent 4 }}
+    annotations:
+    {{ toYaml .Values.annotations | indent 4 }}
```
Comment (zanhsieh) [inline charts/prometheus-stackdriver-exporter/templates/deployment.yaml:8]:
```suggestion
    {{- with .Values.annotations }}
    annotations:
    {{- toYaml .Values.annotations | indent 4 }}
    {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by zanhsieh — APPROVED** [review]

**Review by rpahli — APPROVED** [review]

**Comment by sboschman** [comment]
Seems chart version 4.6.1 is broken, b/c:
```
    {{- with .Values.annotations }}
    annotations:
      {{- toYaml .Values.annotations | indent 6 }}
    {{- end }}
```

```
Error: template: prometheus-stackdriver-exporter/charts/prometheus-stackdriver-exporter/templates/deployment.yaml:9:24: executing "prometheus-stackdriver-exporter/charts/prometheus-stackdriver-exporter/templates/deployment.yaml" at <.Values.annotations>: nil pointer evaluating interface {}.annotations
```

`with` changes the context if I am not mistaken, so the `toYaml` will fail. As `.Values` should come from the root context, e.g. `$.Values.annotations` inside the `with` or simply `{{- toYaml . | indent 6 }}` should work.

**Comment by bc-bishwash** [comment]
This is causing errors, with default values file:
```Error: Error: template: prometheus-stackdriver-exporter/templates/deployment.yaml:9:24: executing "prometheus-stackdriver-exporter/templates/deployment.yaml" at <.Values.annotations>: nil pointer evaluating interface {}.annotations
```

File: charts/prometheus-stackdriver-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: {{ template "stackdriver-exporter.fullname" . }}
   labels:
     {{- include "stackdriver-exporter.labels" . | indent 4 }}
+    {{- with .Values.annotations }}
```
Comment (devops-pump-games) [inline charts/prometheus-stackdriver-exporter/templates/deployment.yaml:7]: I'm not sure, but looks like you should use "{{- if .Values.annotations  }}" to exclude error

```
  Error: template: prometheus-stackdriver-exporter/templates/deployment.yaml:9:24: executing "prometheus-stackdriver-exporter/templates/deployment.yaml" at <.Values.annotations>: nil pointer evaluating interface {}.annotations
  ```

**Pull request merged**